packages/hub/services/validators/scheduled-payment.ts

```diff
@@ -89,23 +89,33 @@ export default class ScheduledPaymentValidator {
       }
     }
 
-    if (scheduledPayment.chainId && !isSupportedChain(scheduledPayment.chainId)) {
-      errors.chainId.push(`chain is not supported`);
-    }
+    if (scheduledPayment.chainId) {
+      if (!isSupportedChain(scheduledPayment.chainId)) {
+        errors.chainId.push(`chain is not supported`);
+      } else {
+        let feeFixedUSD =
+          this.cardpay.getConstantByNetwork('scheduledPaymentFeeFixedUSD', scheduledPayment.chainId) ?? 0;
+        if (Number(scheduledPayment.feeFixedUsd) < feeFixedUSD) {
+          errors.feeFixedUsd.push(`fee USD must be greater than or equal ${feeFixedUSD}`);
+        }
 
-    if (scheduledPayment.chainId && isSupportedChain(scheduledPayment.chainId)) {
-      let feeFixedUSD = this.cardpay.getConstantByNetwork('scheduledPaymentFeeFixedUSD', scheduledPayment.chainId) ?? 0;
-      if (Number(scheduledPayment.feeFixedUsd) < feeFixedUSD) {
-        errors.feeFixedUsd.push(`fee USD must be greater than or equal ${feeFixedUSD}`);
-      }
+        let feePercentage =
+          this.cardpay.getConstantByNetwork('scheduledPaymentFeePercentage', scheduledPayment.chainId) ?? 0;
+        let _feePercentage = Number(scheduledPayment.feePercentage);
```

Comment:
I think introducing a _variable is a bit too noisy and unnecessary here. Can't we just use `feePercentage`?

Reply:
updated